pkg/operator/controller/ingress/status.go

```diff
@@ -171,35 +182,108 @@ func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods
 }
 
 // computeIngressAvailableCondition computes the ingress controller's current Available status state
-// by inspecting the Available condition of deployment. The ingresscontroller is only available if
-// the deployment is also available.
-func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
-	for _, cond := range deployment.Status.Conditions {
-		if cond.Type != appsv1.DeploymentAvailable {
+// by inspecting the following:
+// 1) the Available condition of Deployment,
+// 2) the DNSReady condition of the IngressController, and
+// 3) the LoadBalancerReady condition of the IngressController.
+// The ingresscontroller is judged Available only if all 3 conditions are true
+func computeIngressAvailableCondition(conditions []operatorv1.OperatorCondition) operatorv1.OperatorCondition {
+
+	expected := []expectedCondition{
+		{
+			condition: IngressControllerDeploymentAvailableConditionType,
+			status:    operatorv1.ConditionTrue,
+		},
+		{
+			condition: operatorv1.DNSReadyIngressConditionType,
+			status:    operatorv1.ConditionTrue,
+			ifConditionsTrue: []string{
+				operatorv1.LoadBalancerManagedIngressConditionType,
+				operatorv1.LoadBalancerReadyIngressConditionType,
+				operatorv1.DNSManagedIngressConditionType,
+			},
+			gracePeriod: time.Second * 30,
```

Comment:
Did we decide to specify grace periods in `computeIngressAvailableCondition`?  I thought doing so wasn't necessary for `Available` (it's only necessary for `Degraded` to prevent flapping).

Reply:
I will remove.